### PR TITLE
LPS-96584 getting entries filtered rest query implemented

### PR DESCRIPTION
**Pull request by zoltancsaszi**

### CODE REVIEW

**Comment by zoltancsaszi** [comment]
ci:test:sf

**Comment by zoltancsaszi** [comment]
ci:test:change-tracking

**Comment by zoltancsaszi** [comment]
ci:test:sf

File: modules/apps/change-tracking/change-tracking-rest-impl/src/main/java/com/liferay/change/tracking/rest/internal/resource/v1_0/EntryResourceImpl.java [hunk 1]
```diff
@@ -14,17 +14,154 @@
...
+
+		CTCollection ctCollection = _ctCollectionLocalService.fetchCTCollection(
+			GetterUtil.getLong(collectionId));
+
+		if (ctCollection == null) {
+			throw new IllegalArgumentException(
+				"Unable to get change tacking collection " + collectionId);
+		}
+
+		collision = GetterUtil.getBoolean(collision);
```
Comment (nhpatt) [inline modules/apps/change-tracking/change-tracking-rest-impl/src/main/java/com/liferay/change/tracking/rest/internal/resource/v1_0/EntryResourceImpl.java]: Why this? to see if we can move the logic you need inside a boolean converter

File: modules/apps/change-tracking/change-tracking-rest-impl/src/main/java/com/liferay/change/tracking/rest/internal/resource/v1_0/EntryResourceImpl.java [hunk 2]
```diff
@@ -14,17 +14,154 @@
...
+			SearchUtil.getQueryDefinition(CTEntry.class, pagination, sorts);
+
+		if (status != null) {
+			queryDefinition.setStatus(status);
+		}
+		else {
+			queryDefinition.setStatus(WorkflowConstants.STATUS_DRAFT);
+		}
+
+		List<CTEntry> ctEntries = _ctEngineManager.getCTEntries(
```
Comment (nhpatt) [inline modules/apps/change-tracking/change-tracking-rest-impl/src/main/java/com/liferay/change/tracking/rest/internal/resource/v1_0/EntryResourceImpl.java]: Brian likes this inline (basically if it is only used once, not define a variable), like this: https://github.com/brianchandotcom/liferay-portal/blob/b44c2d166c80b6b6f7738f23f9d85ddc244d6417/modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/KnowledgeBaseFolderResourceImpl.java#L74

File: modules/apps/change-tracking/change-tracking-rest-impl/src/main/java/com/liferay/change/tracking/rest/internal/resource/v1_0/EntryResourceImpl.java [hunk 3]
```diff
@@ -14,17 +14,154 @@
...
+
+		List<Entry> entries = _getEntries(ctEntries);
+
+		return Page.of(entries, pagination, totalCount);
+	}
+
+	private List<Entry> _getEntries(List<CTEntry> ctEntries) {
+		Stream<CTEntry> stream = ctEntries.stream();
+
+		return stream.map(
```
Comment (nhpatt) [inline modules/apps/change-tracking/change-tracking-rest-impl/src/main/java/com/liferay/change/tracking/rest/internal/resource/v1_0/EntryResourceImpl.java]: there is a method inside all resources that does this, transform (it's calling TransformUtil)

File: modules/apps/change-tracking/change-tracking-rest-impl/src/main/java/com/liferay/change/tracking/rest/internal/resource/v1_0/EntryResourceImpl.java [hunk 4]
```diff
@@ -14,17 +14,154 @@
...
+			this::_toEntry
+		).collect(
+			Collectors.toList()
+		);
+	}
+
+	private String[] _getFiltersArray(String filterString) {
```
Comment (nhpatt) [inline modules/apps/change-tracking/change-tracking-rest-impl/src/main/java/com/liferay/change/tracking/rest/internal/resource/v1_0/EntryResourceImpl.java]: mmm can you use arrays instead? they are defined as several parameters &classNameId=20123&classNameId=23434 but they are more standard and converted automatically by JAX-RS implementations

File: modules/apps/change-tracking/change-tracking-rest-impl/src/main/java/com/liferay/change/tracking/rest/internal/resource/v1_0/EntryResourceImpl.java [hunk 5]
```diff
@@ -14,17 +14,154 @@
...
+			throw new IllegalArgumentException(
+				"Unable to get change tacking collection " + collectionId);
+		}
+
+		collision = GetterUtil.getBoolean(collision);
+
+		QueryDefinition<CTEntry> queryDefinition =
+			SearchUtil.getQueryDefinition(CTEntry.class, pagination, sorts);
+
+		if (status != null) {
```
Comment (nhpatt) [inline modules/apps/change-tracking/change-tracking-rest-impl/src/main/java/com/liferay/change/tracking/rest/internal/resource/v1_0/EntryResourceImpl.java]: mmm we could support default values but you should hardcode the id in the OpenAPI profile, what do you think?

**Comment by nhpatt** [comment]
Thanks! :D

**Pull request merged**